src/Microsoft.TestPlatform.TestHostProvider/Hosting/DotnetTestHostManager.cs

```diff
@@ -431,8 +431,15 @@ public Task<bool> LaunchTestHostAsync(TestProcessStartInfo testHostStartInfo, Ca
                     EqtTrace.Verbose($"DotnetTestHostmanager: Forcing the search to x64 architecure, IsDefaultTargetArchitecture '{_runsettingHelper.IsDefaultTargetArchitecture}' OS '{_platformEnvironment.OperatingSystem}' framework '{_targetFramework}'");
                 }
 
+                // Check if DOTNET_ROOT resolution should be bypassed.
+                var shouldIgnoreDotnetRoot = _environmentVariableHelper.GetEnvironmentVariable("VSTEST_IGNORE_DOTNET_ROOT");
```

Comment:
you can compress a bit using this way https://github.com/microsoft/vstest/blob/e97c88d4d53142a23c0e846f068cc9066f9699fd/src/Microsoft.TestPlatform.CoreUtilities/FeatureFlag/FeatureFlag.cs#L33

Reply:
Done.